### PR TITLE
Added 'deadLetterQueueName' as configuration property.

### PR DESCRIPTION
**Pull request by donovanmuller**
Instead of using a hardcoded value (`ActiveMQ.DLQ`) for the DLQ queue name, use a configuration property as a basic improvement.

* Remove all hard coded DLQ references
* Pull out JmsConsumerDestination/JmsProducerDestination

### CODE REVIEW

File: spring-cloud-stream-binder-jms-common/src/main/java/org/springframework/cloud/stream/binder/jms/config/JmsBinderConfigurationProperties.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+import org.springframework.boot.context.properties.ConfigurationProperties;
+
+/**
+ * @author Donovan Muller
+ */
+@ConfigurationProperties("spring.cloud.stream.binder.jms")
+public class JmsBinderConfigurationProperties {
+
+    private String deadLetterQueueName = "dlq";
```
Comment (mbogoevici) [inline spring-cloud-stream-binder-jms-common/src/main/java/org/springframework/cloud/stream/binder/jms/config/JmsBinderConfigurationProperties.java]: Should we perhaps use a more specific name for the DLQ ? "Spring.Cloud.Stream.DLQ" or something like that?

Comment (donovanmuller) [inline spring-cloud-stream-binder-jms-common/src/main/java/org/springframework/cloud/stream/binder/jms/config/JmsBinderConfigurationProperties.java]: 👍 Happy with that.

**Comment by garyrussell** [comment]
Hi Donovan,

I was just wondering what was the rationale for making the producer/consumer destinations top level classes?

I agree that they should have been declared `static` since they are not dependent on the outer class, but I don't think they need to be in their own files.

If you can convince us that is necessary, we'll have to do the same in other binders for consistency.

cc/ @sobychacko

**Comment by sobychacko** [comment]
Looks like JMS case is a bit different. Various provisioning implementation might want to leverage on both `JmsConsumerDestination` and `JmsProducerDestination`. When we made the initial changes we only had Activemq provisioner to work with. Now that we have another provisioiner for ibm-mq, they could reuse them there as well. Thats the rationale I think for why this change is made. I don't think we have the same situation in other binders.

**Comment by donovanmuller** [comment]
@garyrussell The reason mentioned by @sobychacko is accurate. See [here](https://github.com/spring-cloud/spring-cloud-stream-binder-ibm-mq/pull/2/files#diff-7798ceb6c01218379e364d0094e9c8d3R87) and [here](https://github.com/spring-cloud/spring-cloud-stream-binder-ibm-mq/pull/2/files#diff-7798ceb6c01218379e364d0094e9c8d3R108) from https://github.com/spring-cloud/spring-cloud-stream-binder-ibm-mq/pull/2 for reference.

**Comment by garyrussell** [comment]
Doh yes, of course; merging with @mbogoevici 's suggested default for the DLQ.

**Comment by garyrussell** [comment]
Changes for review before I merge: https://github.com/garyrussell/spring-cloud-stream-binder-jms/commit/0a81989d80aaade234f101014b57f56603a940d3

@donovanmuller Please also note (in future) to limit commit headlines to 50 chars.

We generally use the commit guidelines from [Pro Git](https://github.com/spring-projects/spring-integration/blob/master/CONTRIBUTING.adoc#a-day-in-the-life-of-a-contributor).

(I will take care of it when squashing my commit during the merge after my changes are reviewed).

Thanks for the contribution!

**Comment by garyrussell** [comment]
Merged as d812f4ef63f30c6c5e1c0245bdec59c877e8a8e3

**Pull request closed without merging**